src/sensor.h

```diff
@@ -90,6 +90,8 @@ namespace librealsense
             return {};
         }
 
+        virtual bool supports_feature( const std::string & feature_id ) const { return false; } // Default implementation, override in derived classes
```

Comment:
Why in sensor_base and not in sensor_interface?

Reply:
Added here because all logic that uses this API uses `sensor_base` descendants, did not need it more general.
`sensor_interface` is pure abstract, I did not want to add a function with default implementation. If I will add as pure virtual other descendants, like `playback_sensor`, will have to implement too.

Reply:
So code in rs.cpp will not be able to ask if a feature is supported, because it deals with `sensor_interface`?
I think if it's part of the definition of a sensor or device, it should go in the interface.
About the abstract nature, even you have a complete class in `feature_interface`... I think we shouldn't worry about pure abstraction, only that we stick to the proper semantics.